### PR TITLE
handle network errors on webview flow

### PR DESCRIPTION
**Pull request by lbalmaceda**
Check for network status before using the webview. Also let the user retry if there was an error. Note: for this to work, we now require ACCESS_NETWORK_STATE permission.

### CODE REVIEW

**Comment by hzalaz** [comment]
Looks good, can we allow this to work when the permission is not added @lbalmaceda ?

**Comment by hzalaz** [comment]
Also let's have a "Oops" view too, for an error we don't understand (design pending)

File: lock/src/main/java/com/auth0/android/lock/provider/WebViewActivity.java [hunk 1]
```diff
@@ -60,15 +75,44 @@ protected void onCreate(Bundle savedInstanceState) {
...
         progressBar = (ProgressBar) findViewById(R.id.com_auth0_lock_progressbar);
+        progressBar.setIndeterminate(true);
+        progressBar.setMax(100);
+        errorView = findViewById(R.id.com_auth0_lock_error_view);
+        errorMessage = (TextView) findViewById(R.id.com_auth0_lock_text);
+        Button retryButton = (Button) findViewById(R.id.com_auth0_lock_retry);
+        retryButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                isShowingError = false;
```
Comment (hzalaz) [inline lock/src/main/java/com/auth0/android/lock/provider/WebViewActivity.java]: Can we replace this by checking the visibility of `errorMessage`?

Comment (lbalmaceda) [inline lock/src/main/java/com/auth0/android/lock/provider/WebViewActivity.java]: sure!

**Pull request merged**